src/components/Base/Input/input.jsx

```diff
@@ -12,31 +13,55 @@ export default class Input extends React.Component {
     disabled: PropTypes.bool,
     icon: PropTypes.string,
     iconSize: PropTypes.number,
-    iconType: PropTypes.string
+    iconType: PropTypes.string,
+    valueChange: PropTypes.func
   };
 
   static defaultProps = {
     className: '',
     icon: '',
     iconType: 'light',
     iconSize: 16,
+    valueChange: _.noop,
+    onChange: _.noop,
     disabled: false
   };
 
+  onChange = event => {
+    const { target } = event;
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+    const name = target.name;
+
+    this.props.valueChange(name, value);
```

Comment:
valueChange is redundant, you only need `onChange`

Reply:
I want a function pass name and it do not affect before